### PR TITLE
Make Oauth2Client#request retry on auth errors as documented

### LINKED ISSUE #42: Buggy comment in oauth2client.js regarding replay on 401/403
The docstring on `OAuth2Client.prototype.request` says:

```
 * Provides a request implementation with OAuth 2.0 flow.
 * If credentials have a refresh_token, in cases of HTTP
 * 401 and 403 responses, it automatically asks for a new
 * access token and replays the unsuccessful request.
```

But I don't see any logic that would retry on 401/403, nor does that seem plausible given that it wouldn't work with stream bodies.  AFAIK, the token is only refreshed *before* making the request.  (Sorry if I missed something.)

### PR DESCRIPTION
**Pull request by murgatroid99**
Fixes #42 and (I think) #67.

### CODE REVIEW

**Comment by tbetbetbe** [comment]
Please add unit tests that verify the fixes.

**Comment by murgatroid99** [comment]
I've added a test for that

**Comment by murgatroid99** [comment]
Ping

File: test/test.jwt.js [hunk 1]
```diff
@@ -267,6 +267,39 @@ describe('JWT auth client', function() {
       });
     });
 
+    it('should refresh token if the server returns 403', function(done) {
```
Comment (tbetbetbe) [inline test/test.jwt.js]: optional: You can test both 401 and 403 by calling the spec in a loop

**Comment by tbetbetbe** [comment]
LGTM; please merge once you've merged/rebased and the travis CI has passed

**Pull request merged**